### ISSUE TITLE
Testcontainers + failsafe maven plugin not working

### ISSUE THREAD

**Issue by jose-ramirez** [testcontainers/testcontainers-java#4278]
Hi there!

I noticed a problem in a project I'm developing, where trying to run integration tests using the failsafe plugin doesn't work, despite the tests classes running to completion when executed individually (through the IDE).

Here's a minimal project showing the problem I'm talking about: https://github.com/jose-ramirez/mvp

The project has two test classes :`ArticleAPITest` and `UserAPITest`, and after running `mvn test`, or `mvn failsafe:integration-test`, I get the fllowing error:

```
...
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 9.455 s - in hello.world.demo.ArticleAPITest
[INFO] Running hello.world.demo.UserAPITest
2021-07-11 16:17:24.699  INFO 20606 --- [           main] o.s.t.c.support.AbstractContextLoader    : Could not detect default resource locations for test class [hello.world.demo.AbstractIntegrationTest]: no resource found for suffixes {-context.xml, Context.groovy}.
2021-07-11 16:17:24.700  INFO 20606 --- [           main] t.c.s.AnnotationConfigContextLoaderUtils : Could not detect default configuration classes for test class [hello.world.demo.AbstractIntegrationTest]: AbstractIntegrationTest does not declare any static, non-private, non-final, nested classes annotated with @Configuration.
2021-07-11 16:17:24.704  INFO 20606 --- [           main] .b.t.c.SpringBootTestContextBootstrapper : Found @SpringBootConfiguration hello.world.demo.DemoApplication for test class hello.world.demo.UserAPITest
2021-07-11 16:17:24.709  INFO 20606 --- [           main] .b.t.c.SpringBootTestContextBootstrapper : Loaded default TestExecutionListener class names from location [META-INF/spring.factories]: [org.springframework.boot.test.mock.mockito.MockitoTestExecutionListener, org.springframework.boot.test.mock.mockito.ResetMocksTestExecutionListener, org.springframework.boot.test.autoconfigure.restdocs.RestDocsTestExecutionListener, org.springframework.boot.test.autoconfigure.web.client.MockRestServiceServerResetTestExecutionListener, org.springframework.boot.test.autoconfigure.web.servlet.MockMvcPrintOnlyOnFailureTestExecutionListener, org.springframework.boot.test.autoconfigure.web.servlet.WebDriverTestExecutionListener, org.springframework.boot.test.autoconfigure.webservices.client.MockWebServiceServerTestExecutionListener, org.springframework.test.context.web.ServletTestExecutionListener, org.springframework.test.context.support.DirtiesContextBeforeModesTestExecutionListener, org.springframework.test.context.event.ApplicationEventsTestExecutionListener, org.springframework.test.context.support.DependencyInjectionTestExecutionListener, org.springframework.test.context.support.DirtiesContextTestExecutionListener, org.springframework.test.context.transaction.TransactionalTestExecutionListener, org.springframework.test.context.jdbc.SqlScriptsTestExecutionListener, org.springframework.test.context.event.EventPublishingTestExecutionListener]
2021-07-11 16:17:24.711  INFO 20606 --- [           main] .b.t.c.SpringBootTestContextBootstrapper : Using TestExecutionListeners: [org.springframework.test.context.web.ServletTestExecutionListener@4a9f0360, org.springframework.test.context.support.DirtiesContextBeforeModesTestExecutionListener@6f12b637, org.springframework.test.context.event.ApplicationEventsTestExecutionListener@eea0b48, org.springframework.boot.test.mock.mockito.MockitoTestExecutionListener@5feb8e9a, org.springframework.boot.test.autoconfigure.SpringBootDependencyInjectionTestExecutionListener@6d5d1204, org.springframework.test.context.support.DirtiesContextTestExecutionListener@1784a296, org.springframework.test.context.transaction.TransactionalTestExecutionListener@49ea4b65, org.springframework.test.context.jdbc.SqlScriptsTestExecutionListener@4af6839c, org.springframework.test.context.event.EventPublishingTestExecutionListener@4f3e65f4, org.springframework.boot.test.mock.mockito.ResetMocksTestExecutionListener@7d1cfe97, org.springframework.boot.test.autoconfigure.restdocs.RestDocsTestExecutionListener@66b62332, org.springframework.boot.test.autoconfigure.web.client.MockRestServiceServerResetTestExecutionListener@1822a61b, org.springframework.boot.test.autoconfigure.web.servlet.MockMvcPrintOnlyOnFailureTestExecutionListener@240f712e, org.springframework.boot.test.autoconfigure.web.servlet.WebDriverTestExecutionListener@3e7ddb7e, org.springframework.boot.test.autoconfigure.webservices.client.MockWebServiceServerTestExecutionListener@6905d030]
2021-07-11 16:17:24.723  INFO 20606 --- [           main] 🐳 [mongo:latest]                        : Creating container for image: mongo:latest
2021-07-11 16:17:24.759  INFO 20606 --- [           main] 🐳 [mongo:latest]                        : Starting container with ID: c809455f33a689217ea124fe33615ee3647e7449afbd6451a57380b8d31efa0a
2021-07-11 16:17:25.224  INFO 20606 --- [           main] 🐳 [mongo:latest]                        : Container mongo:latest is starting: c809455f33a689217ea124fe33615ee3647e7449afbd6451a57380b8d31efa0a
2021-07-11 16:17:26.449  INFO 20606 --- [           main] 🐳 [mongo:latest]                        : Container mongo:latest started in PT1.726842S
2021-07-11 16:17:26.458  INFO 20606 --- [           main] org.mongodb.driver.connection            : Closed connection [connectionId{localValue:3, serverValue:4}] to localhost:49206 because there was a socket exception raised by this connection.
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.985 s <<< FAILURE! - in hello.world.demo.UserAPITest
[ERROR] getsUser  Time elapsed: 0.015 s  <<< ERROR!
org.springframework.dao.DataAccessResourceFailureException: Prematurely reached end of stream; nested exception is com.mongodb.MongoSocketReadException: Prematurely reached end of stream
	at hello.world.demo.UserAPITest.getsUser(UserAPITest.kt:22)
Caused by: com.mongodb.MongoSocketReadException: Prematurely reached end of stream
	at hello.world.demo.UserAPITest.getsUser(UserAPITest.kt:22)
...
```

Basically, the class `ArticleAPITest` runs fine, with tescontainers doing its job, but when it's time for `UserAPITest` to run, a socket error occurs, which is weird, since running each class separately (through the IDE at least) works just fine.

I tried some things, like reading the logs of ryuk's container and mongo's, but found nothing that could indicate an error. Running the tests in full debug mode didn't show much either. I'm still searching the issue list to see if this has happened before, but nothing yet.

Is this problem the result of a configuration error?

Any pointer to the reason for this to occur will be appreciated.

Also, I placed my question here since it looked like the best place to do so; if it isn't, I'll remove it and put it in the right place.

**Comment by kiview** [comment]
Hey @jose-ramirez, might be better to have this as a question in our [discussions](https://github.com/testcontainers/testcontainers-java/discussions).

Still, after having a quick look at your project, there is initially a weird configuration, which might not necessarily related, but can still be fixed. You are mixing the junit-jupiter extension (https://github.com/jose-ramirez/mvp/blob/master/src/test/kotlin/hello/world/demo/AbstractIntegrationTest.kt#L14) with manual container lifecycle (https://github.com/jose-ramirez/mvp/blob/master/src/test/kotlin/hello/world/demo/AbstractIntegrationTest.kt#L33).
I would suggest just using the manual lifecycle for now and removing the annotation.

**Comment by jose-ramirez** [comment]
That mix was the problem, apparently, 'cause after removing the annotation both commands work now :)

Thanks!
